File: php/features.php
<?php
declare(strict_types=1);

// PassantenfrequenzStadtStgallen SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PassantenfrequenzStadtStgallenFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PassantenfrequenzStadtStgallenBaseFeature();
            case "test":
                return new PassantenfrequenzStadtStgallenTestFeature();
            default:
                return new PassantenfrequenzStadtStgallenBaseFeature();
        }
    }
}

File: py/features.py
# PassantenfrequenzStadtStgallen SDK feature factory

from feature.base_feature import PassantenfrequenzStadtStgallenBaseFeature
from feature.test_feature import PassantenfrequenzStadtStgallenTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PassantenfrequenzStadtStgallenBaseFeature(),
        "test": lambda: PassantenfrequenzStadtStgallenTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
